### PR TITLE
153 support called decision

### LINKED ISSUE #153: Support calledDecision for businessRuleTask
__Is your feature request related to a problem? Please describe.__

In Camunda Cloud `1.3.0`, we will support evaluating DMN decisions in Zeebe from BPMN using a business rule task. The Camunda modeler should allow defining the new properties of a business rule task.  

__Describe the solution you'd like__

For a business rule task,
* a user can choose between the implementation as a job worker (the only option for CC version `1.1` and `1.2`) and the evaluation of a DMN decision within Zeebe (that can be the new default option for version `1.3+`)
* if the user selects the evaluation of a DMN decision then the modeler should show the following properties:
  * new property `decisionId` to define the ID of the decision that is evaluated - it has either a static value or an expression
  * new property `resultVariable` to define the variable name of the decision result (the default value could be the same as `decisionId`)
  * no properties like `Binding`, `Tenant Id` or `Map Decision Result` are supported (from Camunda Platform)
  
__Describe alternatives you've considered__

None.

__Additional context__

Sample XML with the new properties:

```
    <bpmn:businessRuleTask id="chooseDish" name="Choose Dish">
      <bpmn:extensionElements>
        <zeebe:calledDecision decisionId="dish" resultVariable="dish" />
      </bpmn:extensionElements>
      <bpmn:incoming>Flow_0ixwjoa</bpmn:incoming>
    </bpmn:businessRuleTask>
```

Related issue in Zeebe: https://github.com/camunda-cloud/zeebe/issues/8060
Child of camunda/camunda-modeler#2524

### PR DESCRIPTION
**Pull request by MaxTru**
calledDecision

Closes #153

### CODE REVIEW

**Comment by MaxTru** [comment]
cc. @saig0 please see the `gif` - please let me know in case UX is not as exptected.
Note that for final test, I will yet provide you a Camunda Modeler artifact to validate the implementation (once ready)

**Comment by andreasgeier** [comment]
I'm wondering why the implementation dropdown is pre-selected and does not contain a `none` option as we do at other places. 

This also has an effect on the data marker which does not appear on the implementation section if the default is used. It looks odd if the `Called decision` section gets marked but not the parent section `Implementation`.

Can we align with other implementation sections and start without a pre-selection? @MaxTru WDYT?

**Comment by saig0** [comment]
@MaxTru in the gif, it looks like that the custom task headers are available if DMN decision is selected. But the task headers belong to the job worker and should only be available if job worker is selected. Please have a look. Thanks!

**Comment by MaxTru** [comment]
> @MaxTru in the gif, it looks like that the custom task headers are available if DMN decision is selected. But the task headers belong to the job worker and should only be available if job worker is selected. Please have a look. Thanks!

I did not know that, will follow up in a seperate issue: https://github.com/bpmn-io/bpmn-properties-panel/issues/162

Are there any other specific properties which should be disabled if we don't have the `JobWorker` implementation?

**Comment by MaxTru** [comment]
> I'm wondering why the implementation dropdown is pre-selected and does not contain a `none` option as we do at other places.
> 
> This also has an effect on the data marker which does not appear on the implementation section if the default is used. It looks odd if the `Called decision` section gets marked but not the parent section `Implementation`.
> 
> Can we align with other implementation sections and start without a pre-selection? @MaxTru WDYT?

There is no `none` option because:
* a businessRuleTask with `none` would not be executeable. So in any case the user has to select something
* `DMN Decision` is the default as specified by the requirement (see @saig0 s issue). 
* => I personally believe that since `DMN Decision` is the default and probably what is used in the vast majority of cases, it is the better experience to have this pre-selected. Otherwise the user always has to do one extra step.

**Comment by saig0** [comment]
> Are there any other specific properties which should be disabled if we don't have the JobWorker implementation?

No. Input and output mappings are also valid for DMN decisions. Only, custom task headers, job type, and job retries should not be available.

**Review by andreasgeier — APPROVED** [review]
@MaxTru and I discussed my comments and we agreed to move forward with the current solution.

Indeed it is a break in the properties panel behavior (if you compare with business rule task for platform). But it is also a valid argument that we don't have the same amount of different implementation options for the cloud yet. And, if DMN is actually the default in most cases, having it pre-selected makes sense.

Long story short: Let's bring it to the cloud panel and find out if we can learn from feedback.

File: src/provider/zeebe/properties/BusinessRuleImplementationProps.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    let extensionElement, extensionElementType;
+
+    if (value === DMN_IMPLEMENTATION_OPTION) {
+      extensionElement = getCalledDecision(element);
+      extensionElementType = 'zeebe:CalledDecision';
+    } else if (value === JOB_WORKER_IMPLEMENTATION_OPTION) {
+      extensionElement = getTaskDefinition(element);
+      extensionElementType = 'zeebe:TaskDefinition';
+    } else {
+      return;
```
Comment (barmac) [inline src/provider/zeebe/properties/BusinessRuleImplementationProps.js:108]:
```suggestion
    }
```

**Comment by barmac** [comment]
I see that when I switch the implementation type, the `zeebe:calledDecision` stays in the extension elements. I can remember you mentioned that you want to perform the cleanup via a behavior. Will this be added to the PR?

File: src/provider/zeebe/properties/BusinessRuleImplementationProps.js (hunk 1 above)
Comment (barmac) [reply src/provider/zeebe/properties/BusinessRuleImplementationProps.js:108]: If there's no third option, we don't need to handle it.

**Comment by MaxTru** [comment]
> I see that when I switch the implementation type, the `zeebe:calledDecision` stays in the extension elements. I can remember you mentioned that you want to perform the cleanup via a behavior. Will this be added to the PR?

This will be a seperate PR in camunda-bpmn-js (also see breakdown in parent epic)

**Comment by barmac** [comment]
It seems my suggestion broke the code 🙈

**Comment by MaxTru** [comment]
> It seems my suggestion broke the code see_no_evil

I didn't see that either :-) 

Corrected now

**Comment by barmac** [comment]
Already approved from my side. Let's merge this as soon as CI succeeds.
